### PR TITLE
Clean up slotting functions

### PR DESCRIPTION
**Pull request by rvl**
### Overview

I am looking at the changes we need to make in #2245 to get the tests to pass.

The slotting module has a few functions that aren't needed any more.

This PR gets rid of them, so that it's easier to understand the code.

### Comments

This branch doesn't conflict with #2245.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Api/Server.hs [hunk 1]
```diff
@@ -1811,12 +1810,6 @@ getNetworkInformation st nl = do
         handlePastHorizonException _ =
             MaybeT (pure Nothing)
 
-    -- Unsafe constructor for the next epoch. Chances to reach the last epoch
-    -- are quite unlikely in this context :)
-    unsafeEpochSucc :: HasCallStack => W.EpochNo -> W.EpochNo
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Api/Server.hs:1816]: 👍

File: lib/core/test/unit/Cardano/Wallet/Primitive/Slotting/Helpers.hs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- | Legacy slotting API functions.
+-- Used as a reference in tests.
+
+module Cardano.Wallet.Primitive.Slotting.Helpers
```
Comment (Anviking) [inline lib/core/test/unit/Cardano/Wallet/Primitive/Slotting/Helpers.hs]: Happy to see this moved to its own module, but is this not more `Legacy` or `Simple` than `Helpers`?

**Review by Anviking — APPROVED** [review]

File: lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs [hunk 3]
```diff
@@ -692,9 +691,8 @@ monitorStakePools tr gp nl DBLayer{..} =
         -- #2196: We need the try. Arguably we shouldn't need it.
         liftIO (try @SomeException (timeInterpreter nl (epochOf currentSlot))) >>= \case
             Left _ -> return ()
-            Right currentEpoch -> do
-                let subtractTwoEpochs = epochPred <=< epochPred
-                forM_ (subtractTwoEpochs currentEpoch) $ \latestRetirementEpoch -> do
+            Right currentEpoch | currentEpoch >= 2 -> do
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs]: Came to think about: isn't there an otherwise case missing now?

Comment (rvl) [reply lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs]: Yes - oops.

**Pull request merged**